src/transport_ofi.c

```diff
@@ -1353,6 +1355,8 @@ static int shmem_transport_ofi_ctx_init(shmem_transport_ctx_t *ctx, int id)
     info->p_info->mode = 0;
     info->p_info->tx_attr->mode = 0;
     info->p_info->rx_attr->mode = 0;
+    info->p_info->tx_attr->caps = 0;
+    info->p_info->rx_attr->caps = 0;
```

Comment:
Not sure about FI_RECV being required on ctx endpoints in general, however the tx_attr->caps suggestion is good-to-go and I think setting rx_attr->caps to FI_RECV is acceptable for OPX as well.

Reply:
Ok good  - this is more of a note/question for SHMEM developers.  Just a bit of background, each OpenSHMEM context (which is initialized here) is just a transmit resource, the fact that any RX capabilities are specified is a little odd, but it may be needed to drive progress on the CQ...

Reply:
Makes sense! Thanks for the info :)